### ISSUE TITLE
Add missing nvidia-smi calls to plugin.

### ISSUE THREAD

**Issue by Dazog** [influxdata/telegraf#5564]
## Feature Request
Add the following metrics to the nvidia-smi plugin that are missing:
nvidia_smi.pci_bandwidth
nvidia_smi.clocks
vidia_smi.encoder_utilization

Opening a feature request kicks off a discussion.

### Use case: [Why is this important (helps with prioritizing requests)]
To make the GPU monitoring more complete.

**Comment by danielnelson** [comment]
Related #5119 #5532

**Comment by Dazog** [comment]
I can help debug these if you need a tester.

I don't have a way to spin builds to teset these calls, but they were pulled directly from the nvidia-smi application.

**Comment by danielnelson** [comment]
I wonder if we ought to move to the XML output, could you run `nvidia-smi -q -u -x --dtd` and attach the output?

**Comment by Dazog** [comment]
[Nvidia.smi.output.txt](https://github.com/influxdata/telegraf/files/2954416/Nvidia.smi.output.txt)


As requested

**Comment by danielnelson** [comment]
I guess we don't want the `-u` flag, could you run it again without it: `nvidia-smi -q -x --dtd`

**Comment by Dazog** [comment]
[nvidia.smi.take2.txt](https://github.com/influxdata/telegraf/files/2954519/nvidia.smi.take2.txt)
 Yep we have datat now.

**Comment by Dazog** [comment]
Attached is a Turing 1660 Gpu to add to this incase the newest gpu's have more api calls.
[nvidia.smi.turing.txt](https://github.com/influxdata/telegraf/files/3024982/nvidia.smi.turing.txt)

**Comment by DirtyCajunRice** [comment]
@danielnelson how we doing? anything else you need from us?

**Comment by danielnelson** [comment]
I think we have collected the important information to where someone could work on this, but I don't personally have this on my schedule right now.  Will probably depend on a community contribution to be completed.

**Comment by DamianRemotr** [comment]
> 
> 
> I think we have collected the important information to where someone could work on this, but I don't personally have this on my schedule right now. Will probably depend on a community contribution to be completed.

@danielnelson  I am working with Tesla M60. I could try adding missing metrix and test the code. I'm not a skilled developer though, so I'd need some mentoring.

**Comment by danielnelson** [comment]
@DamianRemotr Some of this data could probably be added without switching to the XML format, it might be worth doing this first, since it only requires modifying a few strings in the plugin and updating the tests.  Switching over to the XML format would be a larger overhaul of the plugin, not a massive amount of work but it would be a bit harder.  Will try to help out in either case as much as I can.

**Comment by DirtyCajunRice** [comment]
@danielnelson the csv output is severely limited. there are significantly more metrics available via xml

**Comment by DamianRemotr** [comment]
@danielnelson I'll look at adding some code this weekend and return with eventual questions on Monday. Can I catch you on IRC?

**Comment by danielnelson** [comment]
> Can I catch you on IRC?

No, but you can open a draft PR and we can discuss there.  I might be a bit less available next week due to [InfluxDays](https://influxdays.com/san-francisco-2019/agenda-sf-2019/) and related activities.

**Comment by danielnelson** [comment]
The plugin has been updated to use the XML output in #6639, this should make supporting these new fields easier to implement.

**Comment by reimda** [comment]
It's been a while since this has had attention. @DamianRemotr were you able to put some code together to add the fields? Are you able to put it up in a PR? Thanks!

**Comment by kuriosity121** [comment]
Hi,

we have several NVIDIA Tesla P4 and Tesla T4 GPUs. Those cards are used by remote workers. The P4 are part of a Citrix Hypervisor setup and are used as vGPUs on Windows Virtual Machines. One card is for example devided into 8 virtual parts. Under Windows you can see this card as "GRID P4-1Q".

It would also be nice if you could add information about the current license status of the GPUs. Yes, graphics cards need to be licensed today. If the license is not available performance drops to a minimum. Therefore it would be nice to monitor the current license state.

I will upload a tesla_p4.xml and a tesla_t4.xml for you into the testdata folder. The tesla_p4.xml contains the specific part for the license information which looks like this:

		<vgpu_software_licensed_product>
			<licensed_product_name>NVIDIA RTX Virtual Workstation</licensed_product_name>
			<license_status>Licensed (Expiry: 2022-9-22 6:57:30 GMT)</license_status>
		</vgpu_software_licensed_product>

Can you please add the necessary parts into the plugin?

**Comment by kuriosity121** [comment]
Sorry for the two pull requests. Never worked with git before. I allready noticed, that there might have been an easier way to provide you with the files. But I hope this is OK now.

**Comment by BrentonPoke** [comment]
The encoder_stats_average_fps metric doesn't seem to be working in windows 11 anymore. This is my output.
[output.txt](https://github.com/influxdata/telegraf/files/10975907/output.txt)
